cni/pkg/nodeagent/net_linux_test.go

```diff
@@ -0,0 +1,208 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nodeagent
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/netip"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"sync"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/containernetworking/plugins/pkg/ns"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"istio.io/istio/cni/pkg/ipset"
+	"istio.io/istio/cni/pkg/iptables"
+	"istio.io/istio/pkg/monitoring/monitortest"
+)
+
+// To run this locally, do:
+//   ZTUNNEL_PATH=/path/to/ztunnel go test -v -exec "sudo -E" .
+
+func TestServerAndZtunIntegration(t *testing.T) {
```

Comment:
It does need root. Integ test folder is definitely a better spot. This was mostly for quick local dev testing without spinning up the entire Istio integ suite.

Though it is somewhat redundant with both the other unit tests and with the ambient integ suite, so maybe we just remove it or put it in an `examples` folder.

Reply:
Also it hasn't been running in our internal CI because of these addl requirements and so hasn't been part of our quality gate - as mentioned other tests cover all of this code now - I think @yuval-k used it for local testing at first. Inclined to drop it.